apps/google-analytics-4/frontend/src/components/config-screen/map-account-property/MapAccountPropertySection.tsx

```diff
@@ -0,0 +1,86 @@
+import { useState, useEffect } from 'react';
+import { Stack, Box, Subheading, Paragraph, Select, Spinner } from '@contentful/f36-components';
+import { AccountSummariesType, FlattenedPropertiesType } from 'types';
+import useKeyService from 'hooks/useKeyService';
+import { AppExtensionSDK } from '@contentful/app-sdk';
+import { useSDK } from '@contentful/react-apps-toolkit';
+
+interface Props {
+  accountsSummaries: AccountSummariesType[];
+}
+
+export default function MapAccountPropertySection(props: Props) {
+  const { accountsSummaries } = props;
+  const { handlePropertyChange } = useKeyService({});
+  const sdk = useSDK<AppExtensionSDK>();
+
+  const [selectedPropertyId, setSelectedPropertyId] = useState<string>('');
+  const [flattenedProperties, setFlattenedProperties] = useState<FlattenedPropertiesType[]>([]);
+  const [loadingProperties, setLoadingProperties] = useState<boolean>(true);
+  const [loadingParameters, setLoadingParameters] = useState<boolean>(true);
+
+  useEffect(() => {
+    const fetchParametersFromSdk = async () => {
+      const parameters = await sdk.app.getParameters();
+      if (parameters?.savedPropertyId) setSelectedPropertyId(parameters.savedPropertyId);
+      setLoadingParameters(false);
+    };
+
+    fetchParametersFromSdk();
+  }, [sdk]);
+
+  useEffect(() => {
+    const _flattenedProperties = [] as FlattenedPropertiesType[];
+    accountsSummaries.forEach((accountSummary) => {
+      accountSummary.propertySummaries.forEach((propertySummary) => {
+        _flattenedProperties.push({
+          propertyId: propertySummary.property,
+          propertyName: propertySummary.displayName,
+          accountName: accountSummary.displayName,
+        });
+      });
+    });
+
+    setFlattenedProperties(_flattenedProperties);
+    setLoadingProperties(false);
+  }, [accountsSummaries]);
+
+  const handleSelectionChange = (e: any) => {
+    const _selectedPropertyId = e.target.value;
+    setSelectedPropertyId(_selectedPropertyId);
+    handlePropertyChange(_selectedPropertyId);
+  };
+
+  return (
+    <Stack spacing="spacingL" marginBottom="none" flexDirection="column" alignItems="flex-start">
+      <Box>
+        <Subheading>Configuration</Subheading>
+        <Paragraph>Choose your Account and associated Property</Paragraph>
+        {!loadingProperties &&
+        !loadingParameters &&
+        accountsSummaries.length &&
+        flattenedProperties ? (
```

Comment:
I think this is a little bit of a grey area, but I personally think when render logic has this many moving pieces it's worth pulling out into its own method for readability. It would also make it easier to progressively display loading state depending on the timing of these requisite objects.

Reply:
you mean something like this?
```
  const shouldRenderDropdown = () => {
    return (
      !loadingProperties && !loadingParameters && accountsSummaries.length && flattenedProperties
    );
  };

  return (
    <Stack spacing="spacingL" marginBottom="none" flexDirection="column" alignItems="flex-start">
      <Box>
        <Subheading>Configuration</Subheading>
        <Paragraph>Choose your Account and associated Property</Paragraph>
        {shouldRenderDropdown() ? (
          <Select
```

Reply:
Yeah that's one option that I think could work fine, another would be something like a `renderDropdown` that we would call in the general render, that could handle everything from rendering the dropdown on the happy path, or things like spinners / error messages for other states. 

Right tool for the job right? what you proposed looks great and I think makes the presentational aspect of this way better 👍